docs/ldml/tr35-keyboards.md

```diff
@@ -1388,15 +1400,18 @@ _Attribute:_ `modifier` (required for `hardware`)
 > - `ctrlR`
 > - `shift`
 >
-> Note that `alt` is sometimes referred to as _opt_ or _option_.
+> Note that `alt` in this specification is referred to on some platforms as "opt" or "option".
 >
 > Left- and right- side modifiers (such as `"altL ctrlR"` or `"altL altR"`) should not be used together in a single `modifier` attribute value.
 >
-> Left- and right- side modifiers (such as `"altL ctrlR"` or `"altL altR"`) should not be used together in a single `modifier` attribute value.
-
+>
+> Matching is affected by the set of available layers. For example, if the `shift` and `altL` keys are held down, `<layer id="A" modifiers="altL"/>` will match if there are no other layers which include `altL`. However, if there is also a `<layer id="B" modifiers="altL shift"/>`, then layer `B` will be selected because it is a more exact match.
```

Comment:
Specify the exact algorithm that determines closest match.

Reply:
I don't recall us discussing this and I am not sure it is a good idea. I would prefer if it was no match, no output. That's easy to understand and easy to implement. Otherwise it will get messy, what if there is `shift` only layer as well as `altL` only layer? Would you match `caps+shift` to either of them?  (If needed, we can introduce an attribute refering to a base layer to simplify reusing layers for different modifiers.)

Reply:
perhaps explicit match is a better policy then. We already said in committee that we weren't going to reuse layers at present.

Reply:
Great catch. I'm with you @miloush. Disappointed that I missed this in my review!

We need to be careful with toggle keys (e.g. caps/num/scroll on US keyboards). Num Lock and Scroll Lock are currently excluded as available modifiers but conceptually the spec supports this, and if we ended up with another toggle key in future it could break keyboards if we are not careful.

Reply:
Also note that the example @srl295 gave was easily achieved in the descriptive LDML using `altL altL+shift` (where space worked like _or_ and + like _and_). I am a bit concered that this is not allowed in the prescriptive one, where space means _and_. And since there is no fallback/base mechanism, people need to repeat the whole layer for each combination of modifiers. It's not a tragedy, just checking that it is intentional change.

Reply:
It was an attempt to simplify. Could be too simple, though.

Reply:
@macchiati and all, I'd like to take the modifier logic into a separate PR. So I'd propose to merge this section as-is and have a followon PR (keeping this ticket open) to sort out the modifier(s).